code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -605,8 +605,17 @@
 	id = "iron"
 	description = "Pure iron is a metal."
 	reagent_state = SOLID
+
 	color = "#C8A5DC" // rgb: 200, 165, 220
 
+/datum/reagent/iron/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(!istype(M, /mob/living))
+		return
+	if(M.mind && M.mind.demoninfo && M.mind.demoninfo.banetype == BANEIRON)
+		if(holder && holder.chem_temp < 100) // COLD iron.
+			M.reagents.add_reagent("toxin", reac_volume)
+	..()
+
```

Comment:
All these chem checks are pretty snowflake as well. Maybe make the demon its own mutant race (subtype of human) and check the chems there?